lib/util/pragma.js

```diff
@@ -53,7 +53,16 @@ function getFromContext(context) {
   return pragma;
 }
 
+function getComponentRegExpFromContext(context, pragma) {
+  if (context.settings.react && context.settings.react.component) {
+    return new RegExp(context.settings.react.component);
+  }
```

Comment:
i don't think this is a good solution. the appropriate way to pretend a class inherits from React.Component is https://github.com/yannickcr/eslint-plugin-react/issues/1220#issuecomment-303617161